File: articles/hope-dread-and-the-politics-of-the-future/c/disciplined_hope_score.c
#include <stdio.h>

double disciplined_hope(double hope, double agency, double trust, double institution, double accountability, double repair, double fatigue, double polarization) {
    return 0.18 * hope + 0.18 * agency + 0.16 * trust + 0.16 * institution
        + 0.14 * accountability + 0.12 * repair - 0.04 * fatigue - 0.02 * polarization;
}

int main(void) {
    printf("Reparative Imagination disciplined_hope: %.4f\n",
           disciplined_hope(0.86, 0.84, 0.76, 0.80, 0.88, 0.92, 0.28, 0.30));
    return 0;
}

File: articles/hope-dread-and-the-politics-of-the-future/julia/disciplined_hope_score.jl
# Julia standard-library example for disciplined hope scoring.

profiles = Dict(
    "Disciplined Hope Democracy" => [0.78, 0.82, 0.72, 0.78, 0.84, 0.76, 0.30, 0.34],
    "Apocalyptic Paralysis" => [0.18, 0.22, 0.20, 0.24, 0.28, 0.20, 0.82, 0.60],
    "Reparative Imagination" => [0.86, 0.84, 0.76, 0.80, 0.88, 0.92, 0.28, 0.30]
)

function disciplined_hope(v)
    # v = hope agency trust institution accountability repair fatigue polarization
    0.18*v[1] + 0.18*v[2] + 0.16*v[3] + 0.16*v[4] + 0.14*v[5] +
    0.12*v[6] - 0.04*v[7] - 0.02*v[8]
end

println("scenario,disciplined_hope")
for (name, values) in sort(collect(profiles))
    println("$(name),$(round(disciplined_hope(values), digits=4))")
end
